examples/watering/main.c

```diff
@@ -0,0 +1,219 @@
+#include <stdio.h>
+#include <string.h>
+#include <espressif/esp_wifi.h>
+#include <espressif/esp_sta.h>
+#include <espressif/esp_system.h>
+#include <esp/uart.h>
+#include <esp8266.h>
+#include <etstimer.h>
+#include <esplibs/libmain.h>
+#include <FreeRTOS.h>
+#include <task.h>
+
+#include <homekit/homekit.h>
+#include <homekit/characteristics.h>
+#include "wifi.h"
+
+
+#define MAX_SERVICES 10
+
+const uint8_t valve_gpios[4] = {
+  1, 2, 3, 4
+};
+const size_t valve_count = sizeof(valve_gpios) / sizeof(*valve_gpios);
+
+homekit_accessory_t *accessories[2];
+
+typedef struct {
+    int number;
+    uint8_t gpio;
+    homekit_service_t *service;
+    TaskHandle_t task;
+} gate;
```

Comment:
Why is this called `gate`? Seems like `valve` is a more appropriate name. Also, a style nitpick: usually struct typedefs are easier to recognize if they have a `_t` suffix (e.g. `valve_t`). That also frees `valve` identifier so you can use to name a variable.

Reply:
I renamed it to `valve_t` and `valves`. I have to admit `gate` was very confusing. 😄